### PR TITLE
Refactor: Centralize optional argument parsing for graph functions

### LINKED ISSUE #56: Refactor optional argument (weight, orientation) parsing
During the last wave of PRs, I had to redo the signatures of several functions (including ST_GraphAnalysis, ST_ShortestPathLength) to parse graph orientations (e.g., 'directed - edge_orientation_column'). I ended up with several methods (called parseStringArguments) doing the same thing. I need to put these all in one place to avoid duplicate code.

### PR DESCRIPTION
**Pull request by agouge**
ST_GraphAnalysis, ST_ShortestPathLength and ST_Accessibility all
contained parseOptionalArguments methods. This puts them all in the
GraphFunctionParser.

I add several unit tests to make sure everything is working as expected.

Fixes #56.

### CODE REVIEW

File: src/main/java/org/gdms/gdmstopology/parse/GraphFunctionParser.java [hunk 1]
```diff
@@ -48,102 +56,278 @@
...
+     *
+     * @param v String
+     */
+    private void checkForIllegalCharacters(String v) {
+        Matcher m = Pattern.compile("[^_0-9A-Za-z]").matcher(v);
+        String illegalCharacters = "";
+        while (m.find()) {
+            illegalCharacters += "\"" + m.group() + "\", ";
+        }
+        if (!illegalCharacters.equals("")) {
```
Comment (agueganno) [inline src/main/java/org/gdms/gdmstopology/parse/GraphFunctionParser.java]: I think you may use "isEmpty()" here.

Comment (agouge) [inline src/main/java/org/gdms/gdmstopology/parse/GraphFunctionParser.java]: good catch

**Comment by agueganno** [comment]
Looks pretty good. The choice you make for parsing parameters is unusual but coherent. It may be a bit more documented in the class javadoc, but you say explicitly you use "contains" rather than "equals" or "equalsIgnoreCase" in the methods' javadoc.

**Comment by agouge** [comment]
Anyone care to merge?
@ebocher

**Comment by agueganno** [comment]
Yeah, sry.

**Comment by agouge** [comment]
Thanks

**Pull request merged**